Boolean equality expression transpiles to an invalid query in Django + PostgreSQL

**Describe the bug**
Expressions involving an equality operator and a boolean value fail with `No function matches the given name and argument types. You might need to add explicit type casts.`. This seems to be related with the Django transpiler, and a PostgreSQL database.

**To Reproduce**
Here's a test case that reproduces this behavior:
```python
import pytest
from django.contrib.auth.models import User
from scim2_filter_parser.transpilers.django_q_object import get_query


@pytest.mark.django_db
def test_filter_with_bool_fails() -> None:
    user = User(username='dummy_username', is_active=True)
    user.save()
    query = get_query('active eq true', {('active', None, None): 'is_active'})

    assert list(User.objects.filter(query)) == [user]
```

The test fails with: 
```
(stack trace omited)
E           psycopg.errors.UndefinedFunction: function upper(boolean) does not exist
E           LINE 1: ...ser" WHERE UPPER("auth_user"."is_active"::text) = UPPER(true...
E                                                                        ^
E           HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
```

**Environment**:
Python version: 3.11.7
Django version: 4.2.12
scim-filter-parser version: 0.5.1

**Additional context**
Looking at the source code, I see that `scim2_filter_parser.transpilers.django_q_object.Transpiler.lookup_op` ([source](https://github.com/15five/scim2-filter-parser/blob/master/src/scim2_filter_parser/transpilers/django_q_object.py#L220)) will always transpile `eq` into `iexact`, which then seems to generate a `UPPER(field::text) = UPPER(value)`.

It looks like the `ILIKE` -> `UPPER` change in Django comes from [this ticket](https://code.djangoproject.com/ticket/3575). One could argue that this is a Django issue, though with Django we can choose to use `field__exact=True` instead of `field__iexact=True`, whereas here we rely on the `Transpiler` to do that for us.

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -1,6 +1,10 @@
 CHANGE LOG
 ==========
 
+0.6.0
+----
+- Use "exact" on equality checks against boolean values. #53
+
 0.5.1
 -----
 - Add support for Python 3.12
@@ -101,4 +105,3 @@ match rows now!
 -----
 
 - attr_map keys now control which SQL expressions are present in output of transpiler
-

--- a/src/scim2_filter_parser/transpilers/django_q_object.py
+++ b/src/scim2_filter_parser/transpilers/django_q_object.py
@@ -231,10 +231,10 @@ class Transpiler(ast.NodeTransformer):
         if not op:
             raise ValueError(f"Unknown Django op {op_code}")
 
-        if isinstance(comp_value, bool) and op == 'iexact':
+        if isinstance(comp_value, bool) and op == "iexact":
             # Use "exact" for boolean values, as certain DB drivers (e.g., Postgres) will transpile
             #  "<field> iexact true/false" to "UPPER(field::text) = UPPER(true/false), which fails.
             #  UPPER requires a string.
-            return 'exact'
+            return "exact"
 
         return op or node_value

--- a/src/scim2_filter_parser/transpilers/django_q_object.py
+++ b/src/scim2_filter_parser/transpilers/django_q_object.py
@@ -172,8 +172,6 @@ class Transpiler(ast.NodeTransformer):
         return query
 
     def visit_AttrExprValue(self, node):
-        op = self.lookup_op(node.value)
-
         if node.comp_value:
             # There is a comp_value, so visit node and build SQL.
             # prep item_id to be a str replacement placeholder
@@ -181,6 +179,7 @@ class Transpiler(ast.NodeTransformer):
         else:
             value = None
 
+        op = self.lookup_op(node.value, value)
         if op == "isnull":  # __isnull=False
             value = False
 
@@ -213,7 +212,7 @@ class Transpiler(ast.NodeTransformer):
             return node.value
 
     @staticmethod
-    def lookup_op(node_value):
+    def lookup_op(node_value, comp_value):
         op_code = node_value.lower()
 
         op = {
@@ -231,5 +230,11 @@ class Transpiler(ast.NodeTransformer):
 
         if not op:
             raise ValueError(f"Unknown Django op {op_code}")
+
+        if isinstance(comp_value, bool) and op == 'iexact':
+            # Use "exact" for boolean values, as certain DB drivers (e.g., Postgres) will transpile
+            #  "<field> iexact true/false" to "UPPER(field::text) = UPPER(true/false), which fails.
+            #  UPPER requires a string.
+            return 'exact'
 
         return op or node_value

--- a/tests/test_transpiler_django.py
+++ b/tests/test_transpiler_django.py
@@ -18,6 +18,7 @@ class TestRFCExamples(TestCase):
         ("meta", "lastModified", None): "meta.lastmodified",
         ("ims", "type", None): "ims.type",
         ("ims", "value", None): "ims.value",
+        ("active", None, None): "is_active",
     }
 
     def assert_q(self, input, expected):
@@ -55,12 +56,12 @@ class TestRFCExamples(TestCase):
 
     def test_username_eq_true(self):
         scim = 'userName eq "true"'
-        django = Q(username__iexact=True)
+        django = Q(username__exact=True)
         self.assert_q(scim, django)
 
     def test_username_eq_false(self):
         scim = 'userName eq "false"'
-        django = Q(username__iexact=False)
+        django = Q(username__exact=False)
         self.assert_q(scim, django)
 
     def test_schema_username_startswith(self):
@@ -147,6 +148,11 @@ class TestRFCExamples(TestCase):
             Q(emails__type__iexact="work") & Q(emails__value__icontains="@example.com")
         ) | (Q(ims__type__iexact="xmpp") & Q(ims__value__icontains="@foo.com"))
 
+        self.assert_q(scim, django)
+
+    def test_active_eq_true(self) -> None:
+        scim = 'active eq true'
+        django = Q(is_active__exact=True)
         self.assert_q(scim, django)
 
 

